lib/particle.js

```diff
@@ -0,0 +1,20 @@
+const { Vec3 } = require("vec3")
+
+module.exports = loader
+
+function loader (registry) {
+  function fromNetwork(packet){
+    return {
+      id: packet.particleId,
+      name: registry.particles[packet.particleId].name,
+      forceRender: packet.longDistance,
+      position: new Vec3(packet.x, packet.y, packet.z),
+      offset: new Vec3(packet.offsetX, packet.offsetY, packet.offsetZ),
+      speed: packet.particleData,
```

Comment:
Did you check the accuracy on supported versions?

Reply:
Checked against MC 1.8, MC 1.8.8, MC 1.13, MC 1.16.3 and MC 1.19, can check more version if required

Reply:
Checked with newly written external test, working against all versions